src/main/java/Deadline.java

```diff
@@ -2,10 +2,22 @@
     protected String by;
 
     public Deadline(String description, String by) {
-        super(description);
+        this(description, by, false);
+    }
+
+    public Deadline(String description, String by, boolean isDone) {
+        super(description, isDone);
         this.by = by;
     }
 
+    /**
+     * Returns deadline of task.
+     * @return Deadline of task.
+     */
+    public String getDeadline() {
+        return this.by;
+    }
+
     @Override
     public String toString() {
         return String.format("[D]%s (by: %s)", super.toString(), this.by);
```

Comment:
Very apt use of format specifiers throughout your project when it comes to string processing and printing. Well done!